src/domain/fiat/display-currency.ts

```diff
@@ -25,5 +26,34 @@ export const majorToMinorUnit = ({
   displayMajorExponent: CurrencyMajorExponent
 }) => Number(Number(amount) * 10 ** displayMajorExponent)
 
-export const usdMajorToMinorUnit = (amount: number | bigint) =>
-  majorToMinorUnit({ amount, displayMajorExponent: MajorExponent.STANDARD })
+export const currencyMajorToMinorUnit = ({
+  amount,
+  displayCurrency,
+}: {
+  amount: number | bigint
+  displayCurrency: DisplayCurrency
+}) => {
+  const displayMajorExponent = getCurrencyMajorExponent(displayCurrency)
+  return majorToMinorUnit({ amount, displayMajorExponent })
+}
+
+export const getCurrencyMajorExponent = (
+  currency: DisplayCurrency,
+): CurrencyMajorExponent => {
+  try {
+    const formatter = new Intl.NumberFormat("en-US", { style: "currency", currency })
+    const { minimumFractionDigits } = formatter.resolvedOptions()
+    switch (minimumFractionDigits) {
+      case 0:
+        return MajorExponent.ZERO
+      case 1:
+        return MajorExponent.ONE
+      case 3:
+        return MajorExponent.THREE
+      default:
+        return MajorExponent.STANDARD
```

Comment:
Do we think it's worth it to do something like this in case something weird comes back from the library?:

```
case 2:
        return MajorExponent.STANDARD
default:
        recordExceptionInCurrentSpan({
            error: new UnexpectedMinimumFractionDigitsError(`${minimumFractionDigits}`),
            level: ErrorLevel.Critical,
        })
        return MajorExponent.STANDARD
```

Reply:
:thinking: domain layer should not depend on service layer...

Reply:
Domain layer shouldn't depend on external dependencies at all. If it does (eg as a wrapper like with the `domain/bitcoin` module) we need to deal with its possible return values explicitly to make it safe for other parts of the code to use.

Even if we don't want to return an exception we should at least add some attribute to make it visible that the library we are using isn't working as expected.